packages/core/admin/utils/get-custom-app-config-file.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const { join } = require('path');
+const fse = require('fs-extra');
+
+/**
+ * Retrieve the custom admin entry file name
+ * @param {String} dir
+ * @param {Boolean} useTypeScript
+ * @returns String
+ */
+const getCustomAppConfigFile = async (dir, useTypeScript) => {
+  const adminSrcPath = join(dir, 'src', 'admin');
+  const files = await fse.readdir(adminSrcPath);
+
+  if (useTypeScript) {
+    return files.find(file => file.match(/app.tsx?$/));
+  }
+
+  return files.find(file => file.match(/app.jsx?$/));
```

Comment:
What do you think about normalizing first and therefore shortening this?

```suggestion
  const appRegex = new RegExp(`app.${useTypeScript ? 't' : 'j'}sx?$`)

  return files.find(file => file.match(appRegex));
```

Also: I think it's not mandatory, but I'd love to see a test for this file.